playgame.py

```diff
@@ -16,40 +18,52 @@
 logging.basicConfig()
 logger = logging.getLogger(__name__)
 
+class InvalidHanabiMoveException(Exception):
+    def __init__(self, message, move, player_id):
+        self.message = message
+        self.move = move
+        self.player_id = player_id
+
 def main(argv):
     args = parse_args()
     prep_logger(args.log_dir, args.verbose, args.log_stderr, len(args.players))
+    args.players = validate_players(args.players)
     if args.is_tournament:
         run_tournament(args)
     else:
         run_one_game(args)
 
+def validate_players(players):
+    return [player for player in players if locate(player) is not None and
+            isinstance(locate(player)(), HanabiPlayer)]
+
 def run_tournament(args):
-    tournament_scores = dict.fromkeys(args.players, 0)
+    tournament_scores = dict.fromkeys(args.players, [])
```

Comment:
Now fixes #3 as well
All scores are stored through the tournament. Afterwards, the mean and variance are computed. The winning scores are ones with the highest average and lowest variance